File: algorithms/java/src/PalindromeNumber.java
/**
 * @author jasper
 * @create 2021-09-29 14:03
 */
public class PalindromeNumber {
    public int reverse(int x) {
        int result = 0;
        while (x != 0) {
            int pop = x % 10;
            x /= 10;
            if (result > Integer.MAX_VALUE / 10 || result < Integer.MIN_VALUE / 10) {
                return 0;
            }
            result = result * 10 + pop;
        }
        return result;
    }

    public boolean isPalindrome(int x) {
        if (x < 0) {
            return false;
        }
        int result = reverse(x);
        return result == x;
    }
}

File: algorithms/python/PalindromeNumber.py
# @author jasper
# @create 2021-09-29 14:09

class Solution:
    def reverse(self, x: int) -> int:
        res = 0
        while x != 0:
            x, pop = divmod(x, 10)
            res = res * 10 + pop
        return res

    def isPalindrome(self, x: int) -> bool:
        if x < 0:
            return False
        res = self.reverse(x)
        return res == x
